diff --git a/jailbreak-cabal.cabal b/jailbreak-cabal.cabal
index d34c616..10be8d4 100644
--- a/jailbreak-cabal.cabal
+++ b/jailbreak-cabal.cabal
@@ -21,5 +21,10 @@ source-repository head
 
 executable jailbreak-cabal
   main-is: Main.hs
-  build-depends: base < 5, Cabal == 3.*
+  build-depends:
+    base < 5,
+    Cabal == 3.*
+  if impl(ghc >= 9.3)
+    build-depends:
+      Cabal-syntax
   default-language: Haskell2010
